src/lib/components/forms/form/form-components/text-area/text-area.component.ts

```diff
@@ -0,0 +1,89 @@
+import { Component, Input, OnChanges, SimpleChanges } from '@angular/core';
+import { TerraPlacementEnum } from '../../../../../helpers';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { TextAreaInterface } from './text-area.interface';
+import { noop } from 'rxjs';
+
+@Component({
+    selector: 'tc-text-area',
+    templateUrl: './text-area.component.html',
+    providers: [
+        {
+            provide: NG_VALUE_ACCESSOR,
+            useExisting: TextAreaComponent,
+            multi: true
+        }
+    ]
+})
+export class TextAreaComponent implements OnChanges, ControlValueAccessor, TextAreaInterface {
+    /** If true, the textarea is not resizeable. Default false. */
+    @Input()
+    public hasFixedHeight: boolean = false;
+
+    /** Sets the initial number of rows. Minimum is four. */
+    @Input()
+    public maxRows: number;
+
+    /** Set the label. */
+    @Input()
+    public name: string;
+
+    /** If true, a * indicates that the value is required. Default false. */
+    @Input()
+    public isRequired: boolean = false;
+
+    /** Set the tooltip. */
+    @Input()
+    public tooltipText: string;
+
+    /** If true, the button will be disabled. Default false. */
+    @Input()
+    public isDisabled: boolean = false;
+
+    /** Set the tooltip placement (bottom, top, left, right). Default top. */
+    @Input()
+    public tooltipPlacement: TerraPlacementEnum = TerraPlacementEnum.TOP;
+
+    /** Set a maximum number of characters allowed. */
+    @Input()
+    public maxLength: number;
+
+    /** Internal model. The value of the input. */
+    public value: string;
+
+    /** Stores a callback function which is executed whenever the input was blurred. */
+    public _onTouchedCallback: () => void = noop;
+    /** Stores a callback function which is executed whenever the value of the input changes. */
+    public _onChangeCallback: (_: string) => void = noop;
+
+    private readonly _defaultMaxRows: number = 4;
+
+    constructor() {
+        this.maxRows = this._defaultMaxRows;
+    }
+
+    public ngOnChanges(changes: SimpleChanges): void {
+        if (changes.hasOwnProperty('maxRows') && changes?.maxRows) {
+            this.maxRows = Math.max(this._defaultMaxRows, changes.maxRows.currentValue);
+        }
+
+        if (changes.hasOwnProperty('hasFixedHeight')) {
+            this.hasFixedHeight = !!changes.hasFixedHeight.currentValue;
+        }
+    }
```

Comment:
Das hier funktioniert in der Form tatsächlich nicht richtig und sollte daher auf setter umgebaut werden

Reply:
passe ich an